chrome/android/java/src/org/chromium/chrome/browser/adblock/AdblockInitializer.java

```
@@ -0,0 +1,140 @@
+    }
+
+    private void initAdblock() {
+        final Context context = ContextUtils.getApplicationContext();
+        final String basePath = context.getDir(
+                AdblockEngine.BASE_PATH_DIRECTORY,
+                Context.MODE_PRIVATE)
+                .getAbsolutePath();
+
+        final long ptr = AdblockBridge.getInstance().getIsolateProviderNativePtr();
+        AdblockHelper.get()
+                .init(context, basePath, true, AdblockHelper.PREFERENCE_NAME)
+                .addEngineCreatedListener(this)
+                .useV8IsolateProvider(ptr);
+
+        AdblockHelper.get().getProvider().retain(true);
+    }
+
+    @UiThread
+    private void beInitialized(final AdblockEngine engine) {
+        final long ptr = engine.getFilterEngine().getNativePtr();
+        AdblockBridge.getInstance().setFilterEngineNativePtr(ptr);
+        mInitializationState = mInitializedState;
+        for (final OnAdblockInitListener listener : mInitListeners) {
+            listener.onAdblockInit();
+        }
+        mInitListeners.clear();
+    }
+
+    public interface OnAdblockInitListener {
+        void onAdblockInit();
+    }
+
+    private abstract class InitializationState {
+
+        void init(final ChromeBrowserInitializer initializer) {
+        }
+
+        abstract void addAdblockInitListener(final OnAdblockInitListener listener);
+
+    }
+
+    private final class UnitializedState extends InitializationState {
+
+        @Override
+        void init(ChromeBrowserInitializer initializer) {
+            final BrowserParts parts = new EmptyBrowserParts() {
+                @Override
+                public void finishNativeInitialization() {
+                    initAdblock();
+                }
+            };
+
+            initializer.handlePreNativeStartup(parts);
+            try {
+                initializer.handlePostNativeStartup(true, parts);
+            } catch (final ProcessInitException ignored) {
+            }
```

are we sure we can safely ignore exceptions?